soc/arc/snps_nsim/soc.c

```diff
@@ -12,11 +12,9 @@
  *
  */
 
-#include <zephyr/device.h>
 #include <zephyr/init.h>
-#include "soc.h"
+#include <zephyr/arch/arc/v2/arc_connect.h>
 
-#ifdef CONFIG_SMP
```

Comment:
This doesn't look right, the IDU (interrupt dispatch unit) is only available/supported for SMP configurations.
Why was the ifdef removed?

Reply:
The code has been changed so that this file is only compiled if `CONFIG_SMP=y`. As it was before, the file was always included in the build but had no code when `CONFIG_SMP=n`. See https://github.com/zephyrproject-rtos/zephyr/pull/46585/commits/cf250a2d017e060943121badab8d439d25909576

Reply:
that still does not look right IMO, we probably should leave it as is. Why would generic soc.c be dependent on CONFIG_SMP, soc.c is there for many things, not only enabling SMP. I can see this apply to many other files enable a specific feature, but here I would consider this over-optimization and a trap if someone adds something to soc.c for reasons  other than SMP.

Reply:
> that still does not look right IMO, we probably should leave it as is. Why would generic soc.c be dependent on CONFIG_SMP, soc.c is there for many things, not only enabling SMP. I can see this apply to many other files enable a specific feature, but here I would consider this over-optimization and a trap if someone adds something to soc.c for reasons  other than SMP.

Some socs have more than one C file, here soc.c is not a generic source, so it can likely be renamed to smp.c to avoid confusions. C files becoming empty under certain conditions and still being fed to the compiler isn't a great practice, it just increases the number of files to parse for nothing. cc: @tejlmand

Reply:
Indeed, let's keep this as it is.

Reply:
Agreed, I missed the condition was moved to the CMakeList.txt.

Reply:
still not convinced and, the code in the file is what soc.c is for, it happens to be all SMP specific, fine, does not mean we need to change the name of the file, especially for this type of simulation platforms where code come and goes and soc.c is for this exact puropse.

if we are looking for "performance" gains, I would be looking elsewhere leaving this the way it is.

Reply:
I don't have a strong opinion for one or the other, is there a convention used in other SoCs that we can align with? Consistency over different SoCs would be least confusing and minimizing future problems/mistakes on code updates/additions. For this specific 'SoC' I don't expect we'll add many SoC specific initializations in the future, as this is a CPU-focused simulator.

Reply:
There's probably a similar example on certain ARM SoCs that configure the MMU regions. They have a `mmu_regions.c` that is included only if `CONFIG_ARM_MMU=y`. I think the pattern here is similar. You can actually grep for `zephyr_sources_ifdef` in `soc/`, I see many other examples like for PM. But `soc/` is likely not the best place to look for consistency (see <soc.h>).

Can please maintainers (platform, build system) decide then? I would rather not block progress because of personal preferences here.

Reply:
Well, the main contribution of this PR is the soc.h cleanup. If you keep the soc.c SMP stuff like it was before, I think the PR will be quickly approved. If you feel strong about the SMP init part, you can create an issue for it and/or a separate PR.